rh-hideout/pokeemerald-expansion
# Add AUTO_SCROLL_TEXT and NUM_FRAMES_AUTO_SCROLL_DELAY

# Description
![Birch's intro automatically progresses without button presses](https://files.catbox.moe/5c4ip6.gif)

* Adds `AUTO_SCROLL_TEXT`. If TRUE, text will automatically scroll to the next line after NUM_FRAMES_AUTO_SCROLL_DELAY.

## Details

### Usage

#### `AUTO_SCROLL_TEXT`
As the game normally prints a string in a message box, the game will wait for a button press to process. When `AUTO_SCROLL_TEXT` is active, after `NUM_FRAMES_AUTO_SCROLL_DELAY` is passed, the next line will start printing, even without pressing A.

#### `NUM_FRAMES_AUTO_SCROLL_DELAY`
This is originally defined in vanilla as 49 frames, but can be changed. The lower this number, the faster text will scroll.

# Testing
## Clean Branch
You can recreate this branch by applying a patch or pulling the repo. From a clean version of expansion's upcoming, you can either:

### Patch
`wget https://files.catbox.moe/efv772.patch -O scroll.patch ; git apply scroll.patch ; rm scroll.patch`

### Repo
`git remote add psf-expansion https://github.com/PokemonSanFran/pokeemerald-expansion/ ; git pull psf-expansion autoScrollText`

## Manual Tests
After replicating the branch, to recreate my testing environment, you can either directly download the debug script, or manually create the changes.

### Download
#### `TRUE`
`wget https://files.catbox.moe/oas8nt.h -O include/config/general.h`
#### `FALSE`
`wget https://files.catbox.moe/r7g105.h -O include/config/general.h`

### Manual Testing
* Modify `include/config/general.h` to the desired values
* Compile
* Start new game
* Observe Birch's speech without button presses

## Verified Scenarios
All videos show:
* Player starting a new game and seeing what happens after 30 seconds.

### TRUE
https://github.com/user-attachments/assets/17f2abee-8f2f-479c-b2a7-28a97756771c
### FALSE
https://github.com/user-attachments/assets/7902e1a5-dfca-4753-9565-75eaee445621
# **People who collaborated with me in this PR**
This was originally [written by](<https://www.pokecommunity.com/threads/simple-modifications-directory.416647/page-11#post-10266389>) by Mashanny. Give them all the credit.
# Discord Contact Info

I am `pkmnsnfrn` on Discord.
